kubernetes/llama-stack/README.md

```diff
@@ -0,0 +1,141 @@
+# Llamastack
+
+To run llamastack on OpenShift, we can utlize the kubernetes svc address along with providing a configmap for the unique values we want to test.
+
+
+## Building Llamastack
+
+If you need to use a specific branch or even the main branch of the upstream Llamastack repository in your container image, you can run the following steps:
+
+```
+git clone git@github.com:meta-llama/llama-stack.git
+cd llama-stack
+
+# Create the venv to install llamastack locally
+python -m venv venv
+source venv/bin/activate
+pip install -U .
+
+export CONTAINER_BINARY = podman
+USE_COPY_NOT_MOUNT=true LLAMA_STACK_DIR=. llama stack build --template remote-vllm --image-type container
+
+# tag the local container image and push to quay
+podman tag localhost/distribution-remote-vllm:dev quay.io/redhat-et/llama:2-27-2025
+podman push quay.io/redhat-et/llama:2-27-2025
+```
+
+## Building Llamastack with Milvus (in-line)
+
+If you need to build Llamastack to use Milvus as the default in-line vector db provider in your container image, you can run the following steps:
+
+```
+git clone git@github.com:meta-llama/llama-stack.git
+cd llama-stack
+
+# Create the venv to install llamastack locally
+python -m venv venv
+source venv/bin/activate
+pip install -U .
+```
+
+Edit the `build.yaml` in the `llama_stack/template/remote-vllm/build.yaml` to update the `vector_io` provider field and `image_type` field as shown below:
+
+```
+  providers:
+    vector_io:
+    - inline::milvus
+image_type: container
+```
+
+Now, we can build the container.
+
+```
+export CONTAINER_BINARY = podman
+USE_COPY_NOT_MOUNT=true LLAMA_STACK_DIR=. llama stack build --config llama_stack/templates/remote-vllm/build.yaml --image-type container --image-name remote-vllm-milvus
+```
+
+Once the image is built successfully you can push it to quay:
+
+```
+podman tag localhost/remote-vllm-milvus:<version tag> quay.io/<quay user name or org name>/<image name>
+podman push quay.io/<quay user name or org/<image name>
+```
+
+Update the [`deployment.yaml`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/kubernetes/llama-stack/deployment.yaml#L71) file using the image generated above.
+
+## Configmap
+
+A predefined [`configmap`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/kubernetes/llama-stack/configmap.yaml) exists with entirely too many options, but is a placeholder for what can be added. Create the configmap defining your unique settings. Then create the CM. The [`template.yaml`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/kubernetes/llama-stack/template.yaml) file does not need to be modified. It contains the jinja chat template.
+
+```
+oc create -f llama-stack/configmap.yaml
+oc create -f llama-stack/template.yaml
+```
+
+## Deployment
+
+To create the Llamastack deployment:
+
+```
+oc create -f llama-stack/deployment.yaml
```

Comment:
There's also a PVC required for persisting llama stack deployment files and the yaml for that exists in the repo. We should mention it here.